### PR TITLE
support local src mounting with subpaths

### PR DESCRIPTION
**Pull request by moritzraho**
implements #23

### CODE REVIEW

File: src/invoker.js [hunk 1]
```diff
@@ -127,6 +130,8 @@ class OpenWhiskInvoker {
             this.sourceDir = this.sourcePath;
             this.sourceFile = "";
         }
+        // sourceRoot is cwd unless a build path has been defined in which case it is buildPathRoot or buildPath directory
+        this.sourceRoot = this.hasBuildPath && (this.buildPathRoot || this.sourceDir) || this.cwd
```
[the remark this answers is not in the document]
Comment (moritzraho) [reply src/invoker.js]: @alexkli thanks for the review. :)

- IIUC in #23 we discussed that if buildPath is defined we keep the old way a.k.a parent folder if buildPath is a file or buildPath if buidPath is a dir

- Maybe I am missing smthg but as per the condition `this.sourceDir` should be reachable if `--build-path` is defined (`this.hasBuildPath===true`) and there is no `--build-path-root` option set (`this.buildPathRoot===false`).

[the remark this answers is not in the document]
Comment (moritzraho) [reply src/invoker.js]: but yeah I guess I could get rid of requiring the buildPath option if buildPathRoot is set, will check tmw

[the remark this answers is not in the document]
Comment (alexkli) [reply src/invoker.js]: Ah, I thought `hasBuildPath` is based off of `buildPathRoot`.

I think this expression could be simplified to
```
= this.buildPath && (this.buildPathRoot || this.sourceDir) || this.cwd
```

Anyway, it's getting hard to follow, let me cover this through tests.

[the remark this answers is not in the document]
Comment (moritzraho) [reply src/invoker.js]: ok let me try to simplify the logic a bit

[the remark this answers is not in the document]
Comment (moritzraho) [reply src/invoker.js]: @alexkli I think it's ok to use `this.hasBuildPath` because there is no `this.buildPath`
because of constructor `this.sourcePath = options.buildPath || options.sourcePath;`.

We can of course change this logic and expose both`buildPath` and `sourcePath` as two different entities to the developer of the kind wrapper, but as only buildPath is mounted into the container I don't think it's a good idea

I agree that things are not very clear but not sure how to simplify best
for reference and to recap, as of now we need:
`sourcePath/Dir` as watch path for live reload
`cwd` as mount path if buildPath is not defined
`buildPath/Dir` as mount path if buildRootPath is not defined
`buidRootPath` as mount path if buildPath is defined

**Comment by alexkli** [comment]
@moritzraho I finally got the test running. This has 3 test failures: https://app.circleci.com/jobs/github/adobe/wskdebug/154/tests

I can continue to look at it tomorrow.

**Comment by alexkli** [comment]
I think the problem is that it does not handle the case where no source file is given at all (e.g. `wskdebug myaction`) and throws this error then:

```
Error: [source-path] or --build-path must point to the action javascript source file, it cannot be a folder.
```

**Comment by moritzraho** [comment]
Thanks for the adding those !
@alexkli I've added a test and done the fixed for the failing ones

File: src/kinds/nodejs/nodejs.js [hunk 2]
```diff
@@ -30,12 +31,12 @@ module.exports = {
...
     // return extra docker arguments such as mounting the source path
     dockerArgs: function(invoker) {
-        if (invoker.sourceDir) {
+        if (invoker.sourcePath) {
             if (!invoker.sourceFile) {
                 throw new Error("[source-path] or --build-path must point to the action javascript source file, it cannot be a folder.");
             }
 
-            return `-v "${invoker.sourceDir}:${CODE_MOUNT}"`;
+            return `-v "${invoker.sourceRoot}:${CODE_MOUNT}"`;
```
Comment (alexkli) [inline src/kinds/nodejs/nodejs.js]: Using `invoker.sourcePath` above vs. `invoker.sourceRoot` seems suboptimal - from this code's perspective, `sourcePath` could be set, but `sourceRoot` not. Maybe we can streamline it in `invoker.js` a bit to just have `sourceDir`.

**Comment by alexkli** [comment]
New PR on top of this is in #35.
